### DISCUSSION TITLE
Install instructions don't work and Vite hangs forever on init

### DISCUSSION THREAD

**Discussion by burlesona** [ElMassimo/vite_ruby#19] [Q&A] [answered]
Hey, @ElMassimo, not sure if I even know where to start to turn this into a proper issue report (hence I started a discussion instead), but basically Vite Rails doesn't seem to work. Two main issues I've run into so far:

1. If I run bin/vite dev it doesn't start the dev server, it just prints the commands as if I hadn't entered a command at all.
2. After updating my rails app to use Vite, the server tries to process a request and it does say "building with vite" but it just hangs there forever and never builds.

I know this is a crappy kind of bug report, but I would love to get onto something faster than web pack, and this seems to be a fairly new thing that I imagine you're still working the kinks out of? If that's right and you'd be up for it, I'd be happy to hop on a video chat sometime and we could work on it together to figure it out.

Cheers!
- Andrew

**Comment by cangencler** [comment]
Same problem here. 

rails version: `5.2.2`
node version `14.13.0`

**Comment by ElMassimo** [comment]
Hey there!

I've been using the integrations for the past three weeks without hiccups. The project is fully tested, and there are [live examples](https://github.com/ElMassimo/pingcrm-vite) that build as expected on CIs and Heroku.

Please run `bin/vite info` and post the result, as well as the node version and other relevant npm packages.

Is `npx` available in your system? Are `vite` and `vite-plugin-ruby` installed in the project and which versions? The installation script should have added them, but a user recently mentioned that they were not installed. If you can paste the output of running `bundle exec vite install`, that would help me to understand what might cause it and fix it for future installations.

To confirm whether it's environment-related, please run [this example](https://github.com/ElMassimo/pingcrm-vite), or the Rails example in this repo.

- If the examples work for you, then perhaps you can share an example repo where the problem occurs.
- If the examples don't work, then it's environment-related, and we should investigate what's happening inside the Ruby process. If you run `bundle open dry-cli`, and add a debugger statement, it should be possible to figure out why the dev command is not found. Same by running `bundle open vite_ruby`. For debugging, I like to use pry.

Sorry for the long response, but I won't have computer access until next Wednesday, so I'd like to give you a few options to try in the meantime.

**Reply by ElMassimo** [reply]
Fixed installation of JS packages in #23, and made some changes to prevent silent failures.

**Comment by ElMassimo** [comment, marked as answer]
Based on the logs provided in https://github.com/ElMassimo/vite_ruby/issues/22, it's most likely that you have experienced a bug in the installation process, which caused `vite` and `vite-plugin-ruby` not to be installed.

Add both packages to your development dependencies and install them, and the problem should go away.

**Reply by burlesona** [reply]
Hey @ElMassimo, this did "fix the problem" in a way, thanks!

It also uncovered a new thing. Possibly this is worth a new discussion thread, just let me know.

I use Tailwind, and bundling with Vite was unfortunately just as slow as Webpacker (12 seconds for any change to a CSS file). Do you know anything about Vite and Tailwind - is Tailwind just a dog?

There were also several issues where Tailwind dependencies (postcss plugins) had to be explicitly added to the devDependencies list, whereas with Webpacker those were loaded implicitly after having Tailwind as a dependency. Not sure if you've run into that before.

Any thoughts? Thanks for being responsive and engaged here, I know it's kind of a PITA doing OSS work and I'm grateful you're willing to take the time to chat about it :)

**Reply by ElMassimo** [reply]
Now that you mention it, I've been contributing to a new project to address that pain point.

It's called [Windi CSS](https://github.com/windicss/windicss), it has a [Vite plugin](https://github.com/windicss/vite-plugin-windicss), and I've made a [new documentation website](http://windicss.netlify.app/) for it. Also created a [Discord server](https://discord.gg/mUQkK8mq), where you can ask questions.

You should be able to get [insane load times in comparison](https://twitter.com/antfu7/status/1361398324587163648).

Adam Wathan (creator of Tailwind CSS) was impressed with Windi CSS, and is currently implementing [something similar](https://twitter.com/adamwathan/status/1363638620209446921) himself after learning about it.

@burlesona Give it a try, and let me know how it goes, I think you won't be disappointed.
